sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -407,80 +366,25 @@ export class MessageSender extends LinkEntity<AwaitableSender> {
   }
 
   /**
-   * Will reconnect the sender link if necessary.
+   * Closes the rhea link.
+   * To be called when connection is disconnected, onAmqpClose and onSessionClose events.
    * @param {AmqpError | Error} [senderError] The sender error if any.
    * @returns {Promise<void>} Promise<void>.
    */
   async onDetached(senderError?: AmqpError | Error): Promise<void> {
+    if (senderError) {
```

Comment:
If `onDetached()` gets called with no error, we end up not calling `closeLink()`. Is that intentional?

Reply:
Woah! Wait.. `closeLink` does get called. `if` from line 375 ends at line 383, and the `closeLink` gets called irrespective of the error being present.

Reply:
My bad